syntax = "proto3";
package iscPacket;

message ServerReg
{
	enum ServerType {
		LOGIN = 0;
		CHAR = 1;
		NODE = 2;
		MAP_MASTER = 3;
		MAP_WORKER = 4;
	}

	ServerType type = 1;
	string name = 2;
	string addr = 3;
	int32 port = 4;
	int32 accright = 5;
}

message ServerStatus
{
	enum Status {
		ONLINE = 0;
		FAILED = 1;
		FULL = 2;
		INVALID = 3;
		MAINT = 4;
	}
	Status status = 1;
	int32 userPercent = 2;
}

message ClientTransfer
{
	int32 userid = 1;
	int32 charid = 2;
	int32 serverid = 3;
}
